forge-game/src/main/java/forge/game/card/CardFactoryUtil.java

```diff
@@ -3964,6 +3996,15 @@ else if ("Artifact".equals(t))
             String effect = "Mode$ CantBlockBy | ValidAttacker$ Creature.Self | ValidBlocker$ Creature.withoutFlying+withoutReach | Secondary$ True" +
                     " | Description$ Flying (" + inst.getReminderText() + ")";
             inst.addStaticAbility(StaticAbility.create(effect, state.getCard(), state, intrinsic));
+        } else if (keyword.startsWith("Harmonize")) {
+            String raiseEffect = "Mode$ OptionalCost | ValidCard$ Card.Self | ValidSA$ Spell.Harmonize | Secondary$ True"
+                    + " | Cost$ tapXType<1/Creature.powerEQHarmonize/creature for Harmonize> | EffectZone$ All "
+                    + " | Description$ Harmonize (" + inst.getReminderText() + ")";
+            inst.addStaticAbility(StaticAbility.create(raiseEffect, state.getCard(), state, intrinsic));
+            String reduceEffect = "Mode$ ReduceCost | ValidCard$ Card.Self | Type$ Spell.Harmonize | Secondary$ True"
```

Comment:
@Hanmac
We still need to find a cleaner way to connect this
I assume right now if a card has two instances of Harmonize it would apply twice?

Reply:
I have a plan for this, but it takes some bigger changes.

For the Card/SpellAbility remembering which optional cost was paid, I want to add an extra list especially for the generic cost to hold the Static Ability and/or keyword that did it.

Then in the ReduceCost, you could do something like "is linked" or something.

The problem is that OptionalCostValue can't be the object for this, because that one still needs to be serialize-able.

OR maybe better, what if we change Optional Cost Static? Add a param like "ReduceCost"
So while CostAdjustment, add the Static Abilities that Increased or just only the Optional Cost ones, and then for reducing, it already has the needed parameter 
(That can help clean up the Defiler cards #1828 )

Reply:
Is this ready for now? Or should I try to do something with the Defilers?

Reply:
@tool4ever what instead of the Announce, we move the Cost increase to addExtraKeywordCost like with Casualty?

Reply:
it's still experimental and I forgot split card as final test: they fail the KW check 😢

Reply:
Should we still move it to `addExtraKeywordCost`?

Reply:
it might be cleaner, will try after dinner